### PR TITLE
util/log: replace gen.sh by a go template generator

### PR DESCRIPTION
**Pull request by knz**
This prevents build problems due to folk not using a `bash` that's
modern enough.

Release note: None

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1544,24 +1544,21 @@ docs/generated/redact_safe.md:
 	  sed -E -e 's/^([^:]*):[0-9]+:.*redact\.RegisterSafeType\((.*)\).*/\1 | \`\2\`/g' >>$@.tmp || { rm -f $@.tmp; exit 1; }
 	@mv -f $@.tmp $@
 
-docs/generated/logging.md: pkg/util/log/gen.sh pkg/util/log/logpb/log.proto
-	bash $< logging.md >$@.tmp || { rm -f $@.tmp; exit 1; }
+docs/generated/logging.md: pkg/util/log/gen.go pkg/util/log/logpb/log.proto
+	$(GO) run $^ logging.md $@.tmp || { rm -f $@.tmp; exit 1; }
```
Comment (otan) [inline Makefile:1548]: may need $(GOMODVENDOR) flags

File: pkg/util/log/gen.go [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+		Severities []info
+		Channels   []info
+	}{sevs, chans}); err != nil {
+		return err
+	}
+
+	// If we are generating a .go file, do a pass of gofmt.
+	newBytes := src.Bytes()
+	if strings.HasSuffix(tmplName, ".go") {
+		newBytes, err = format.Source(newBytes)
```
Comment (otan) [inline pkg/util/log/gen.go:69]: why not crlfmt ;) ?

File: pkg/util/log/gen.go [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+	return nil
+}
+
+type info struct {
+	RawComment string
+	Comment    string
+	PComment   string
+	Name       string
+	NAME       string
```
Comment (otan) [inline pkg/util/log/gen.go:97]: why not `NameUpper`? 

I think you cal also use `call strings.Lower "bob"` in a template.

**Review by otan — APPROVED** [review]

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57589)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal and @otan)

___
*[Makefile, line 1548 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57589#-MNjrJmzpdM71DE-Srnl-r1-1548:-MNwp3kH47w1c4cX9sFc:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7451e458a335b875651a6d12b056766d12728cc/Makefile#L1548)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

may need $(GOMODVENDOR) flags
</blockquote></details>

Done.
___
*[pkg/util/log/gen.go, line 69 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57589#-MNjrJmzpdM71DE-Srns-r1-69:-MNwp4KnETfKmJRq9h0K:bqntp9n) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7451e458a335b875651a6d12b056766d12728cc/pkg/util/log/gen.go#L69)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

why not crlfmt ;) ?
</blockquote></details>

This code does not need it.

(Also crlfmt is not built to be embedded alas)
___
*[pkg/util/log/gen.go, line 97 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/57589#-MNjrJmzpdM71DE-Srns-r1-97:-MNwrQn9-BWBfaLwJwxj:b-erv8ma) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7451e458a335b875651a6d12b056766d12728cc/pkg/util/log/gen.go#L97)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

why not `NameUpper`? 

I think you call also use `call strings.Lower "bob"` in a template, but this looks used often enough
</blockquote></details>

Yes for the same reason .NAME is shorter than .NameUpper. I cannot use `.name` because templates only work with exported names (:sob:)

**Comment by knz** [comment]
bors r=otan

**Pull request merged**